TestCases/compliance-level-3/0083-feel-unicode/0083-feel-unicode.dmn

```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<definitions namespace="http://www.montera.com.au/spec/DMN/0083-feel-unicode" name="0083-feel-unicode" id="_i9fboPUUEeesLuP4RHs4vA" xmlns="http://www.omg.org/spec/DMN/20180521/MODEL/" xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/" xmlns:dmndi="http://www.omg.org/spec/DMN/20180521/DMNDI/" xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <description>FEEL unicode tests</description>
+
+    <decision name="decision_001" id="_decision_001">
+        <variable name="decision_001"/>
+        <literalExpression>
+            <text>string length("\u0009")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_002" id="_decision_002">
+        <variable name="decision_002"/>
+        <literalExpression>
+            <text>string length("\\u0009")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_003" id="_decision_003">
+        <variable name="decision_003"/>
+        <literalExpression>
+            <!-- horse emoji -->
+            <text>string length("\uD83D\uDCA9")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_003_a" id="_decision_003_a">
+        <variable name="decision_003_a"/>
+        <literalExpression>
+            <!-- horse emoji lowercase -->
+            <text>string length("\ud83d\udca9")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_004" id="_decision_004">
+        <variable name="decision_004"/>
+        <literalExpression>
+            <!-- horse + grinning face emoji -->
+            <text>string length("\ud83d\udc0e\uD83D\uDE00")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_004_a" id="_decision_004_a">
+        <variable name="decision_004_a"/>
+        <literalExpression>
+            <!-- horse + grinning face emoji -->
+            <text>string length("🐎😀")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_005" id="_decision_005">
+        <variable name="decision_005"/>
+        <literalExpression>
+            <!-- horse + grinning face emoji -->
+            <text>contains("\ud83d\udc0e\uD83D\uDE00", "\uD83D\uDE00")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_005_a" id="_decision_005_a">
+        <variable name="decision_005_a"/>
+        <literalExpression>
+            <!-- horse + grinning face emoji -->
+            <text>contains("\ud83d\udc0e\uD83D\uDE00", "😀")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="decision_006" id="_decision_006">
+        <variable name="decision_006"/>
+        <literalExpression>
+            <text>{🐎: "bar"}</text>
```

Comment:
FEEL grammar rule for `61. key = name | string literal ;` taking the `name` path;
which in turn is `27. name = name start , { name part | additional name symbols } ;`
which in turn is `30. name start char = ...`
I do not believe the horse-emoji can be used in here (without making it within a string literal double quote start/end)

Reply:
Rule 30 on page 110 says:

name start char = "?" | [A-Z] | "_" | [a-z] | [\uC0-\uD6] | [\uD8-\uF6] | [\uF8-\u2FF] | [\u370-\u37D] | [\u37F-\u1FFF] | [\u200C-\u200D] | [\u2070-\u218F] | [\u2C00-\u2FEF] | [\u3001-\uD7FF] | [\uF900-\uFDCF] | [\uFDF0-\uFFFD] | [\u10000-\uEFFFF] ;

The character is 1F40E which clearly falls into the last range mentioned in rule 30.

Seems to me that this is explicitly allowed by the spec unless I am misunderstanding something about the spec.

Reply:
Thanks @agilepro , @tarilabs.  I just did the same calc, plus doubled checked the lexer code and I believe it is a valid start of name char.

Reply:
Worth mentioning here that this is not effected by the \uXXXX issue because in this case the character is specified in UTF-8.  There are no ambiguities or outstanding questions with UTF-8 directly in the source text.   In this case, the character is clearly specified in the text, and the spec says it is allowed.   Matteo, please look at it again.

Reply:
will do.
On other hand this test suite is affected by the problem with the `string length` which is still not addressed ;)